src/main.rs

```diff
@@ -56,13 +58,28 @@ fn initialize_js(input_path: &PathBuf,
         .arg(path);
     }
 
+    command.arg("--internal-wizening-mode");
+
+    let source = fs::read_to_string(input_path)?;
+    let lits = regex::find_literals(&source);
+    let mut temp = NamedTempFile::new()?;
+
+    let path = if !lits.is_empty() {
+        writeln!(temp, "{}", &source)?;
+        regex::precompile(&lits, &mut temp)?;
+        &temp.path()
+    } else {
+        input_path.as_path()
+    };
+
+    let js_file = fs::File::open(path)
+    .with_context(|| format!("failed to open JS file: {}", path.display()))?;
+    command.arg(path).stdin(js_file);
+
     let status = command
-    .arg("--internal-wizening-mode")
-    .arg(input_path)
-    .arg(output_path)
-    .stdin(js_file)
-    .status()
-    .expect("Failed to invoke wizening mode");
+            .arg(output_path)
+            .status()
+            .expect("Failed to invoke wizening mode");
```

Comment:
Is this the indentation that rustfmt would use? (I haven't really gotten used to rustfmt's rules.) Related thought: sometime we ought to add rustfmt alongside clang-format in the format-check CI job.

Reply:
Just to follow up, this is how rustfmt formatted the file.